patches/@open-wc+scoped-elements+2.0.1.patch

```diff
@@ -0,0 +1,113 @@
+ import { dedupeMixin } from '@open-wc/dedupe-mixin';
+ import { adoptStyles } from '@lit/reactive-element/css-tag.js';
+ 
+@@ -10,6 +9,9 @@ import { adoptStyles } from '@lit/reactive-element/css-tag.js';
+  * @typedef {import('@lit/reactive-element').CSSResultOrNative} CSSResultOrNative
+  */
+ 
++// @ts-ignore
++const supportsScopedRegistry = !!ShadowRoot.prototype.createElement;
++
+ /**
+  * @template {import('./types').Constructor<HTMLElement>} T
+  * @param {T} superclass
+@@ -86,7 +88,6 @@ const ScopedElementsMixinImplementation = superclass =>
+       this.constructor.__registry = registry;
+     }
+ 
+-    /** @override */
+     createRenderRoot() {
+       const {
+         scopedElements,
+@@ -95,11 +96,10 @@ const ScopedElementsMixinImplementation = superclass =>
+       } = /** @type {typeof ScopedElementsHost} */ (this.constructor);
+ 
+       if (!this.registry) {
+-        this.registry = new CustomElementRegistry();
+-
+-        Object.entries(scopedElements).forEach(([tagName, klass]) =>
+-          this.registry.define(tagName, klass),
+-        );
++        this.registry = supportsScopedRegistry ? new CustomElementRegistry() : customElements;
++        for (const [tagName, klass] of Object.entries(scopedElements)) {
++          this.defineScopedElement(tagName, klass);
++        }
+       }
+ 
+       /** @type {ShadowRootInit} */
+@@ -109,16 +109,23 @@ const ScopedElementsMixinImplementation = superclass =>
+         customElements: this.registry,
+       };
+ 
+-      this.renderOptions.creationScope = this.attachShadow(options);
+-
+-      if (this.renderOptions.creationScope instanceof ShadowRoot) {
+-        adoptStyles(this.renderOptions.creationScope, elementStyles);
++      const createdRoot = this.attachShadow(options);
++      if (supportsScopedRegistry) {
++        this.renderOptions.creationScope = createdRoot;
++      }
+ 
+-        this.renderOptions.renderBefore =
+-          this.renderOptions.renderBefore || this.renderOptions.creationScope.firstChild;
++      if (createdRoot instanceof ShadowRoot) {
++        adoptStyles(createdRoot, elementStyles);
++        this.renderOptions.renderBefore = this.renderOptions.renderBefore || createdRoot.firstChild;
+       }
+ 
+-      return this.renderOptions.creationScope;
++      return createdRoot;
++    }
++
++    createScopedElement(tagName) {
++      const root = supportsScopedRegistry ? this.shadowRoot : document;
++      // @ts-ignore polyfill to support createElement on shadowRoot is loaded
++      return root.createElement(tagName);
+     }
+ 
+     /**
+@@ -128,7 +135,24 @@ const ScopedElementsMixinImplementation = superclass =>
+      * @param {typeof HTMLElement} klass
+      */
+     defineScopedElement(tagName, klass) {
+-      return this.registry.get(tagName) || this.registry.define(tagName, klass);
++      const registeredClass = this.registry.get(tagName);
++      if (registeredClass && supportsScopedRegistry === false && registeredClass !== klass) {
```

Comment:
Wouldn't this break already for every instance of LionInput that creates LionValidationFeedback (at instance level and not per constructor?), a.k.a. as soon as we don't load the polyfill ourselves internally in Lion?

Reply:
this is comparing the `class` and the `LionValidationFeedback` will always be the same no matter how many `<lion-validate-feedback>` instances you create...

also I guess at least one test should fail if that would. not be the case

Reply:
Exactly, looked over the `registeredClass !== klass` part :)